File: Swift多线程/NSThread/NSThread/ViewController.m
//
//  ViewController.m
//  NSThread
//
//  Created by 廖佩志 on 2019/8/26.
//  Copyright © 2019 廖佩志. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (strong, nonatomic) NSThread * thread;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)test2 {
    NSLog(@"3");
}
- (void)test {
//    [self test2];
    NSLog(@"2");
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{

    NSThread *thread = [[NSThread alloc] initWithBlock:^{
        NSLog(@"1");
        [[NSRunLoop currentRunLoop] addPort:[NSPort port] forMode: NSRunLoopCommonModes];
        [[NSRunLoop currentRunLoop] run];
//        NSLog(@"zzzzz%@", [NSRunLoop currentRunLoop]);
    }];

    [thread start];

    [self performSelector:@selector(test) onThread: thread withObject:nil waitUntilDone: YES];
    NSLog(@"123");
//    NSLog(@"%@", self.thread);
//    [];
//    [self performSelectorInBackground:@selector(test) withObject:nil];
}

@end

File: Swift卡片式轮播组件/Swift卡片式轮播组件/ViewController.swift
//
//  ViewController.swift
//  Swift卡片式轮播组件
//
//  Created by 廖佩志 on 2019/9/6.
//  Copyright © 2019 廖佩志. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.

        let carouselView = CarouselView(frame: CGRect(x: 0, y: 0, width: view.bounds.width, height: 200))
        carouselView.center = view.center
        carouselView.images = ["插图1.png","插图2.png","插图3.png"]
        view.addSubview(carouselView)
    }
     

}

